hibernate-search-plugins/org.jboss.tools.hibernate.search.runtime.common/src/org/jboss/tools/hibernate/search/runtime/common/AbstractFacadeFactory.java

```diff
@@ -0,0 +1,85 @@
+package org.jboss.tools.hibernate.search.runtime.common;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import org.jboss.tools.hibernate.runtime.common.IFacade;
+import org.jboss.tools.hibernate.runtime.common.Util;
+import org.jboss.tools.hibernate.runtime.spi.ISession;
+import org.jboss.tools.hibernate.runtime.spi.ISessionFactory;
+import org.jboss.tools.hibernate.search.runtime.spi.IAnalyzer;
+import org.jboss.tools.hibernate.search.runtime.spi.IFullTextSession;
+import org.jboss.tools.hibernate.search.runtime.spi.IQueryParser;
+import org.jboss.tools.hibernate.search.runtime.spi.ISearchFactory;
+
+public abstract class AbstractFacadeFactory implements IFacadeFactory {
```

Comment:
hum ok but I think Reflection could have been avoided using an adaptor and loading from "the right" module the needed implementation.

This solution might work but you'll not be able to notice when a new version breaks, on changing some of the method signatures you need.